packages/jest-snapshot/src/index.ts

```diff
@@ -122,22 +122,35 @@ const cleanup = (
   hasteFS: HasteFS,
   update: Config.SnapshotUpdateState,
   snapshotResolver: JestSnapshotResolver,
-) => {
+): {
+  filesRemoved: number;
+  filesRemovedList: Array<string>;
+} => {
   const pattern = '\\.' + EXTENSION + '$';
   const files = hasteFS.matchFiles(pattern);
-  const filesRemoved = files.reduce((acc, snapshotFile) => {
-    if (!fileExists(snapshotResolver.resolveTestPath(snapshotFile), hasteFS)) {
-      if (update === 'all') {
-        fs.unlinkSync(snapshotFile);
+  const {count, list} = files.reduce(
+    (acc: {count: number; list: Array<string>}, snapshotFile) => {
```

Comment:
can we do this instead?

```ts
  const {count, list} = files.reduce<{count: number; list: Array<string>}>(
    (acc, snapshotFile) => {
```

moves the type info from the callback to the `reduce` itself

Reply:
sure